### PR TITLE
CharSequence variant of JsonStringEncoder.quoteAsString

### PR DESCRIPTION
**Pull request by mikaelstaldal**
Fix for https://github.com/FasterXML/jackson-core/issues/265

### CODE REVIEW

File: src/test/java/com/fasterxml/jackson/core/io/TestJsonStringEncoder.java [hunk 1]
```diff
@@ -37,7 +50,24 @@ public void testQuoteLongAsString() throws Exception
         assertEquals(exp, new String(result));
         
     }
-    
+
+    // For [JACKSON-853]
```
Comment (cowtowncoder) [inline src/test/java/com/fasterxml/jackson/core/io/TestJsonStringEncoder.java]: Could you remove this, or change to point to [core#266] (or whatever was id of the issue)?

**Comment by cowtowncoder** [comment]
Looks good, and I'd be happy to merge it in. The only thing before that is to make sure I have an CLA for you -- I don't think I've asked for one yet?

You can find the doc at:

https://github.com/FasterXML/jackson/blob/master/contributor-agreement.pdf

and the usual thing to do is to print, fill & sign, scan and email it to `info` at fasterxml dot com.
If you could do that, I can merge the PR as soon as I have it. Also, this is one time thing: further merges are fine for any and all Jackson projects. Apologies for having to do this, but it's the minimum we need from process perspective (some companies are very strict on ensuring OSS libs they use make sure to get confirmation that all contributions are valid etc).

File: src/test/java/com/fasterxml/jackson/core/io/TestJsonStringEncoder.java (hunk 1 above)
Comment (mikaelstaldal) [inline src/test/java/com/fasterxml/jackson/core/io/TestJsonStringEncoder.java]: OK.

**Comment by cowtowncoder** [comment]
Looks good, got CLA, merging.

**Pull request merged**